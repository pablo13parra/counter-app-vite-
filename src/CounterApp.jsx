import { useState } from 'react';
import PropTypes from 'prop-types';


// const csname = (a, b) => (a + b)

export const CounterApp = ({ value }) => {

    const [counter, setCounter] = useState(value);

    const handleAdd = () => {
        // setCounter(counter + 1);
        setCounter((c) => c + 1);
    }

    const handleSubstract = () => {
        // setCounter(counter - 1);
        setCounter((c) => c - 1);
    }

    const handleReset = () => {
        setCounter(value);
        // setCounter((c) =>  c = value );
    }

    return (
        <>
            <h1>CounterApp</h1>
            <h2>{counter}</h2>
            <button aria-label="btn-inc" onClick={handleAdd}> +1 </button>
            <button aria-label="btn-dec" onClick={handleSubstract}> -1 </button>
            <button aria-label="btn-reset" onClick={handleReset}> Reset </button>
        </>
    )
}

CounterApp.propTypes = {
    value: PropTypes.number,
}