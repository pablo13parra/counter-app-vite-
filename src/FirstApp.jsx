import PropTypes from 'prop-types';

// const csname = (a, b) => (a + b)

export const FirstApp = ({ title, subtitle, name }) => {
    // if (!title) { throw new Error('NO EXISTE EL TITULO, SOS DE MADERA') }

    return (
        <>
            <h1 data-testid="test-title">{title}</h1>
            <p>{subtitle}</p>
            <p>{subtitle}</p>
            <p>{name}</p>
        </>
    )
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string,
    name: PropTypes.string
}

FirstApp.defaultProps = {
    subtitle: 'no hay',
    name: 'xd'
}