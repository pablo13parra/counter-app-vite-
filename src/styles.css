body, html {
    background-color: #000;
    color: #FFF;
    text-align: center;
}

button {
    padding: 15px;
    font-size: 3em;
    margin-right: 5px;
}