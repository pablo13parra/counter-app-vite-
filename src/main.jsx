import React from 'react';
import ReactDOM from 'react-dom/client';
// import { HelloWorldApp } from './HelloWorld';
import { CounterApp } from './CounterApp';
import { FirstApp } from './FirstApp';
import './styles.css';

ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        <CounterApp value={20}/>
        {/* <FirstApp title="Soy Vegeta"/> */}
    </React.StrictMode>
)