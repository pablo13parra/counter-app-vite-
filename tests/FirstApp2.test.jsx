import { render, screen } from '@testing-library/react';
import { FirstApp } from '../src/FirstApp'
describe('pruebas FirstApp', () => {
    const title = 'hi im gokú'
    const subtitle = 'hi im subtitle'
    test('should match with snapshot', () => {
        const { container } = render(<FirstApp title={title} />);
        expect(container).toMatchSnapshot();
    });

    test('should show the message "hi im gokú"', () => {
        render(<FirstApp title={title} />);
        expect(screen.getByText(title)).toBeTruthy();
    });
    test('should show the title on h1 tag', () => {
        render(<FirstApp title={title} />);
        expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain(title);
    });

    test('should send the subtitle by props', () => {
         render(
            <FirstApp
                title={title}
                subtitle={subtitle}
            />
        );
        expect(screen.getAllByText(subtitle).length).toBe(2);
    });

});
