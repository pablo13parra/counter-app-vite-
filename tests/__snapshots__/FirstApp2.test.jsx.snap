// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas FirstApp should match with snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    hi im gok√∫
  </h1>
  <p>
    no hay
  </p>
  <p>
    no hay
  </p>
  <p>
    xd
  </p>
</div>
`;
